src/common/telemetry/with-telemetry.ts

```diff
@@ -0,0 +1,11 @@
+export interface TelemetryOptions {
```

Comment:
@quesabe let's use `type` and split them across their own files.

Reply:
JSII does not support `type` as imports and fails with the following error: `JSII1003: Only string-indexed map types are supported`.